src/internal/deploy/assets/postgres.go

```diff
@@ -380,3 +385,75 @@ EOSQL
 		},
 	}
 }
+
+// PostgresService generates a Service for the pachyderm pg bouncer deployment.
+func PGBouncerService(opts *AssetOpts) *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: objectMeta(pgBouncerName, labels(pgBouncerName), nil, opts.Namespace),
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeNodePort,
+			Selector: map[string]string{
+				"app": pgBouncerName,
+			},
+			Ports: []v1.ServicePort{
+				{
+					Port:     5432,
+					Name:     "client-port",
+					NodePort: opts.PostgresOpts.Port,
+				},
+			},
+		},
+	}
+}
+
+// PGBouncerDeployment
+func PGBouncerDeployment(opts *AssetOpts) *apps.Deployment {
+	return &apps.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: objectMeta(pgBouncerName, labels(pgBouncerName), nil, opts.Namespace),
+		Spec: apps.DeploymentSpec{
+			Replicas: replicas(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels(pgBouncerName),
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: objectMeta(pgBouncerName, labels(pgBouncerName), nil, opts.Namespace),
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  pachdName,
+							Image: pgBouncerImage,
+							Env: []v1.EnvVar{
+								{Name: "DB_USER", Value: postgresUser},
+								{Name: "DB_PASSWORD", Value: postgresPassword},
```

Comment:
Do we need to specify a password? Aren't we using trust auth for everything anyways?

Reply:
I think you need the password to tell it to use password auth as opposed to one of the other auth methods.